splitio/sync/impression.py

```diff
@@ -84,7 +84,7 @@ def __init__(self, impressions_api, impressions_counter):
 
     def synchronize_counters(self):
         """Send impressions from both the failed and new queues."""
-        to_send = self._impressions_counter.pop_all()
+        to_send = self._impressions_manager._strategy._counter.pop_all()
```

Comment:
maybe you can pass counter instead of manager or at least check if impression_manager is not none and strategy is not none and counter is not none